rpc/core/tx.go

```diff
@@ -127,9 +139,55 @@ func TxSearch(
 			Index:    r.Index,
 			TxResult: r.Result,
 			Tx:       r.Tx,
-			Proof:    proof,
+			Proof:    txProof,
 		})
 	}
 
 	return &ctypes.ResultTxSearch{Txs: apiResults, TotalCount: totalCount}, nil
 }
+
+func proveTx(height int64, index uint32) (types.TxProof, error) {
+	var (
+		pTxProof tmproto.TxProof
+		txProof  types.TxProof
+	)
+	rawBlock, err := loadRawBlock(env.BlockStore, height)
+	if err != nil {
+		return txProof, err
```

Comment:
It is perhaps more natural to declare the first return type as a pointer(*types.TxProof), so we can return `nil` here.
I checked the two callsites of this function, and this wouldn't be a problem.